code/datums/gamemode/objectives/target/assassinate.dm

```diff
@@ -1,5 +1,14 @@
+var/list/assassination_objectives = list()
+
 /datum/objective/target/assassinate
 	name = "Assassinate <target>"
+	var/double_agent_completion_id
+	var/syndicate_checked = 0
+
+/datum/objective/target/assassinate/New(var/text,var/auto_target = TRUE, var/mob/user = null)
+	..()
+	double_agent_completion_id = generate_artifact_id()//guarranted to be unique, will let us check for completion when taking pictures
```

Comment:
Are you seriously making a completion ID based on total randomness rather than basing it on the target ? A weakref to the target would be much better.

Reply:
First off as the one who made that proc I assure you it's guaranteed to not have twice the same id per target, AND we need an id that isn't shared by any other assassination objectives, in which case target refs aren't suitable because there might be several objectives with the same target during a round (mostly due to getting a copy of your victim's objective when you kill them, and no we can't just move the objective over to them because we also need to remember that the victim originally had that objective and failed)

Reply:
I assume you meant a weakref to the objective itself?

Reply:
Using weakrefs is indeed way simpler here. Just do that.

For future reference: if you actually need a unique ID system like this... just use a global counter. It's 10x simpler and not just... confusing. Using `generate_artifact_id()` would imply some sort of user-visibility of these IDs which is not the case.